crates/cheatcodes/src/evm.rs

```diff
@@ -224,6 +224,24 @@ impl Cheatcode for resumeGasMeteringCall {
     }
 }
 
+impl Cheatcode for lastCallGasCall {
+    fn apply(&self, state: &mut Cheatcodes) -> Result {
+        let Self {} = self;
+        ensure!(
```

Comment:
it should work without the isolate flag too right, no? It would only change the gas returned as it takes into account the 21k base cost + other costs?

Reply:
You are correct, it does work but, as you indicated, lacks accuracy. I've removed the explicit ensure.
I've now generalized the Solidity test so it gets both tested in non-isolation and isolation mode.